### DISCUSSION TITLE
Sword & Horde - Development Diary #1 - The Game Plan

### DISCUSSION THREAD

**Discussion by RodRitter** [evennia/evennia#2666] [My Evennia Project]
This will be the first of, hopefully many, dev diary entries which will document my learning and planning process for a simple end-to-end MUD adventure. Sword & Hoard is going to be a small scale vertical slice of a game. Think of it as a polished "tutorial" for a larger game, with combat, levelling, trading, etc.

---

## Introducing Sword & Hoard
Sword & Hoard is going to be a rogue-lite, hack & slash, dungeon delver. I'm aiming for the mechanics to be simple to understand, with the potential to add depth later on.

### Single Player First
For this slice of the game, I'll be focussing on making the single-player experience as fun as possible. This means that things like emotes and PvP will be left of the backburner for now.

### The Story Premis
You are a servant for a travelling merchant. One day he makes you an offer to delve and slay the denizens of a cursed ruin. If you gather enough ingredients and exotic creature parts, he tells you that he will set you free of your servitude contract.

_Think of this like a tutorial to a much larger game_

### Combat
Combat will be an interval auto-battle system, where the player can change their "stance" at any time to give them an edge. The results are generated using a d6 system (six-sided dice):  

`Attack (3d6+mod) - Defense (2d6+mod) = Damage`

The player can change their default stance during battles, to add modifiers:

- Aggressive: `+4 Attack Roll / -2 Defence Roll`
- Defensive: `+4 Defense Roll / -2 Attack Roll`
- Balanced: `+1 Attack & +1 Defense`

Players and enemies will have 2 seconds between auto-attacks.

### Equipment & Loot
Weapons and armour will give you a direct modifier to your attack and defend rolls. Jewellery/Accessories will give bonuses to specific stances.

- Weapons - `+1 to +3 Attack Mod`
- Armour - `+1 to +3 Defence Mod`
- Amulet/Ring - `+1 to +3 A/D/B Stance Mod`

### Player Levelling
There will only be 10 levels planned for this slice. Each level, you can choose to increase one of the following:

- `+1 Attack`
- `+1 Defense`
- `+1 Attack: A-Stance`
- `+1 Defense: D-Stance`

### Rooms
The areas will be simple. (Flavour and descriptions can be added later)

- Ominous Ruin Entrance: Merchant is available here
- Ruin (Rooms: 1-20) - Difficulty will scale as you go deeper.

It's going to be a linear path for now.

## Conclusion
And that's the basics for creating a working "tutorial"/demo that can be played from start to finish. Obviously, some details need to be fleshed out as I build the systems.

The next post will be focused on some of the basics, like **room descriptions** and **moving around**

**Comment by Zireael07** [comment]
Why is attack 3d6 and defence 2d6? That's really assymmetric and weird and massively prioritizes the attacker.

**Reply by RodRitter** [reply]
My idea is to have a d6 advantage so combat isn't drawn out. If it were completely equal, there would be a lot  of misses (or 0 dmg hits)

It benefits the attacker, but each combatant gets the same advantage.

I'm not saying I've completely balanced it, but I don't want to spend too much time dwelling over it. It'll be adjusted over time once I can play test.
